rawfile/rawfile_servicer.py

```diff
@@ -212,8 +222,25 @@ def CreateVolume(self, request: csi_pb2.CreateVolumeRequest, context):
                     request.volume_content_source.volume.volume_id,
                 )
             ):
-                # TODO: Create temporary snapshot
-                raise VolumeCloningNotSupported()
+                remove_source_snapshot = True
+                source_meta = metadata(request.volume_content_source.volume.volume_id)
+                current_snapshot = snapshot_manager.list_snapshots(
+                    volume_id=request.volume_content_source.volume.volume_id,
+                    snapshot_name=f"{request.volume_content_source.volume.volume_id}-clone-source",
+                    temporary=True,
+                )
+                if current_snapshot.data and current_snapshot.data[0].ready:
+                    snapshot_id = current_snapshot.data[0].snapshot_id
+                else:
+                    snapshot_id = snapshot_manager.create_snapshot(
```

Comment:
so we create a snapshot during the call here?
and then we clone the snapshot 
and then we delete the snapshot?

Could you please extend the other document with this behavior as well?
Also I wonder, can not simply create the snapshot as the clone? I mean the snapshot *is* the clone right? Without `reflink` this would avoid copying twice?

Reply:
Right, in case of `reflink` yes that makes sense, my case was more for non-`cow`, where you have to deep copy twice.

Reply:
Yeah, But I was just like not making it conditional, It can get complex

Reply:
Yeah we'd need another entity to determine which is the best approach depending on the parameters.
We could consider doing it as another improvement.

Reply:
Yeah, I will merge this for now